Markup.cpp

```diff
@@ -1163,35 +1146,38 @@ int _vscprintf(const char* format, va_list pargs)
 }
 #endif
 
-std::string CMarkup::Format(const char *fmt, ...)
+std::string CMarkup::Format(const char* fmt, ...)
 {
-    std::string retStr;
-    if (NULL != fmt)
+    if (!fmt)
     {
-        va_list marker;
+        return "";  // EARLY EXIT
+    }
 
-        // initialize variable arguments
-        va_start(marker, fmt);
+    va_list marker;
+    va_start(marker, fmt);
+    va_list markerCopy;  // Used for determining buffer size
+    va_copy(markerCopy, marker);
 
-        // Get formatted string length adding one for NULL
-        int len = _vscprintf(fmt, marker) + 1;
+    // Determine buffer size
+#if defined(_MSC_VER)
```

Comment:
See if it's reasonable to replace this entire if-else statement with vsnprintf_s.

Reply:
Removed Windows-specific code.